api/src/main/java/org/sonarsource/scanner/api/internal/OkHttpClientFactory.java

```
@@ -59,8 +60,17 @@ private OkHttpClientFactory() {
   static OkHttpClient create(Logger logger) {
     OkHttpClient.Builder okHttpClientBuilder = new OkHttpClient.Builder();
 
+    int readTimeout = 0;
+    try {
+      if (!System.getProperty(READ_TIMEOUT_SEC_PROPERTY, "").isEmpty()) {
+        readTimeout = Integer.parseInt(System.getProperty(READ_TIMEOUT_SEC_PROPERTY));
+      }
+    } catch (NumberFormatException e) {
```

In case of bad number, I would let the exception throw and fail fast. Silently ignoring it may put the user in a bad situation where he doesn't understand why his setting is not taken into account.

---

Updated.